Logging/DeprecationHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of Ekino New Relic bundle.
+ *
+ * (c) Ekino - Thomas Rabaix <thomas.rabaix@ekino.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ekino\Bundle\NewRelicBundle\Logging;
+
+use Ekino\Bundle\NewRelicBundle\Exception\DeprecationException;
+use Ekino\Bundle\NewRelicBundle\NewRelic\NewRelicInteractorInterface;
+
+class DeprecationHandler
+{
+    private $isRegistered = false;
+    private $interactor;
+
+    public function __construct(NewRelicInteractorInterface $interactor)
+    {
+        $this->interactor = $interactor;
+    }
+
+    public function register(): void
+    {
+        if ($this->isRegistered) {
+            return;
+        }
+        $this->isRegistered = true;
+
+        $prevErrorHandler = set_error_handler(function ($type, $msg, $file, $line, $context = array()) use (&$prevErrorHandler) {
```

Comment:
This looks super cool. But does it really work with `$prevErrorHandler = .... use (&$prevErrorHandler)...` like that?

Reply:
Let me right some tests to prove it :)